File: LC118.java
import java.util.*;

public class question_118 {
    class Solution{
        public List<List<Integer>> generate(int numRows) {
            List<List<Integer>> answer = new ArrayList<List<Integer>>();
            for (int i = 0; i < numRows; i++){
                List<Integer> row = new ArrayList<Integer>();
                row.add(1);
                for(int j = 1; j<=i; j++){
                    row.add(j, (int)((long)(row.get(j-1))*(i-j+1)/j));
                }
                answer.add(row);
            }
            return answer;
        }
    }
}

File: LC118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        answer = list()
        for i in range(numRows):
            row = list()
            for j in range(i+1):
                if j == 0 or j == i:
                    answer.append(1)
                else:
                    row.append(answer[i-1][j-1] + answer[i-1][j])
            answer.append(row)
        return answer